<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

# Exercise 1 (Learning to interrupt) 

In [1]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

# Exercise 2 (Resetting & Restarting the Kernel)

In [2]:
print = 'PRINT'

In [1]:
print('Hello World')

Hello World


# Exercise 3 (Timing Stuff)

In [6]:
%%timeit
squares = [i ** 2 for i in range(10)]

431 ns ± 10 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [7]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i ** 2)

507 ns ± 32.4 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


- The first value (507 ns) is the mean execution time of the cell.
- The second value (±32.4 ns) is the standard deviation of the execution time of the cell.
- The cell is run for 7,000,000 times in total (7 runs and 1,000,000 loops per run).

## Footnotes